From 62eeecfd79a6a401a77f621cb97044af27f01454 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:32 -0400
Subject: [tools] perf: Set event->header.misc to PERF_RECORD_MISC_GUEST_USER if machine is guest

Message-id: <1408953210-26343-205-git-send-email-jolsa@redhat.com>
Patchwork-id: 88894
O-Subject: [PATCH RHEL7.1 BZ1133083 204/322] perf tools: Set event->header.misc to PERF_RECORD_MISC_GUEST_USER if machine is guest.
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 73547aacdcd45cdf8b53245ded9593d7c587501a
Author: Dongsheng Yang <yangds.fnst@cn.fujitsu.com>
Date: Fri Dec 20 15:52:58 2013 -0500

description
===========
When we synthesize the mmap events of user space, if machine is guest,
we should set the event->header.misc to PERF_RECORD_MISC_GUEST_USER,
rather than PERF_RECORD_MISC_USER.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/event.c b/tools/perf/util/event.c
index a61726e..07c0783 100644
--- a/tools/perf/util/event.c
+++ b/tools/perf/util/event.c
@@ -220,7 +220,10 @@ static int perf_event__synthesize_mmap_events(struct perf_tool *tool,
   /*
    * Just like the kernel, see __perf_event_mmap in kernel/perf_event.c
    */
-  event->header.misc = PERF_RECORD_MISC_USER;
+  if (machine__is_host(machine))
+   event->header.misc = PERF_RECORD_MISC_USER;
+  else
+   event->header.misc = PERF_RECORD_MISC_GUEST_USER;
 
   if (prot[2] != 'x') {
    if (!mmap_data || prot[0] != 'r')
-- 
1.7.1